packages/core/src/render3/interfaces/node.ts

```diff
@@ -502,14 +502,25 @@ export interface TNode {
   projection: (TNode|RNode[])[]|number|null;
 
   /**
-   * A collection of all style static values for an element.
+   * A collection of all `style` static values for an element (including from host).
    *
    * This field will be populated if and when:
    *
-   * - There are one or more initial styles on an element (e.g. `<div style="width:200px">`)
+   * - There are one or more initial `style`s on an element (e.g. `<div style="width:200px;">`)
+   * - There are one or more initial `style`s on an directive/component host
+   *   (e.g. `@Directive({host: {style: "width:200px;" } }`)
    */
   styles: string|null;
 
+
+  /**
+   * A collection of all `style` static values for an element excluding host sources.
+   *
+   * Populated when there are one or more initial `style`s on an element
+   * (e.g. `<div style="width:200px;">`)
```

Comment:
Might want to add the reason why this needs to be stored separately from `tNode.styles`, e.g.

```suggestion
   * (e.g. `<div style="width:200px;">`)
   * Must be stored separately from `tNode.styles` to facilitate setting directive 
   * inputs that shadow the `style` property. 
   * If we used `tNode.styles` as is for shadowed inputs, we would feed host styles 
   * back into directives as "inputs". 
   * If we used `tNode.attrs`, we would have to concatenate the attributes on 
   * every template pass. Instead, we process once on first create pass and store here.
   
```